mlflow/types/type_hints.py

```diff
@@ -45,6 +46,44 @@
 }
 
 
+@lru_cache(maxsize=1)
+def no_op_type_hints():
```

Comment:
```suggestion
def unsupported_type_hints_for_signature():
```

"no-op" sounds like it returns special type hints that do not represent anything (like `Any`).

Reply:
`unsupported` is a bit weird as well since we allow them 🤔 Maybe `type_hints_needing_signature`?

Reply:
Oh "unsupported" here means "unsupported for signature inference". I'm not very confident as non-native speaker, but "needing" sounds a bit strange. What about `type_hint_not_signature_inferable`? A few ChatGPT alternatives:

- `type_hint_signature_unsupported`
- `type_hint_signature_inference_disabled`
- `type_hint_no_signature_inference`

Reply:
@serena-ruan it still sound a bit strange to say "type hints need signature", because type hint itself doesn't require anything, and also it's unclear what "need signature" means (does this mean users need to set signature manually?).

Reply:
I'll pick `type_hints_no_signature_inference` as it explains what we do with those type hints 🤔